src/azul/__init__.py

```diff
@@ -501,6 +498,11 @@ class Catalog:
 
         _it_catalog_re: ClassVar[re.Pattern] = re.compile(r'it[\d]+')
 
+        def __attrs_post_init__(self):
+            for plugin_type in ('metadata', 'repository'):
```

Comment:
I think this can simply require equality between the dict keys and the plugin type names. If you want a fancy log message, compute the symmetric difference and list the missing keys and the extra keys in the error message. If you do use symmetric difference, the requirement should be that the symmetric difference is two empty sets.